pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1188,6 +1193,14 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 	return tlsContext, nil
 }
 
+// ApplyECDHCurves applies the ECDH curves from mesh config to UpstreamTlsContext
+// Used for building upstream TLS context for mesh external TLS/mTLS origination
+func ApplyECDHCurves(tlsContext *auth.UpstreamTlsContext, mesh *meshconfig.MeshConfig) {
+	if mesh != nil && mesh.MeshExternal_TLS != nil && len(mesh.MeshExternal_TLS.EcdhCurves) > 0 {
+		tlsContext.CommonTlsContext.TlsParams.EcdhCurves = mesh.MeshExternal_TLS.EcdhCurves
```

Comment:
We should validate the values to prevent sending invalid curves to envoy

Reply:
@hzxuzhonghu since ecdh curves is in Meshconfig, the validation is added while configuring MeshConfig.